src/components/entityset/EntitySetStorage.js

```diff
@@ -0,0 +1,61 @@
+/* @flow */
+import { schema } from 'normalizr';
+import { PropTypes } from 'react';
+import { Permission } from '../../core/permissions/Permission';
+
+/* Type */
+export type Type = {
+  name: string,
+  namespace: string
+};
+
+export const TypePropType = PropTypes.shape({
+  name: PropTypes.string.isRequired,
+  namespace: PropTypes.string.isRequired
+});
+
+/* PropertyType */
+export type PropertyType = {
+  id: string,
```

Comment:
`id` is only optional on write. It's required on read, and `normalizr` needs it for storage. I prefer keeping it required, then treating write as a special case